crates/ruff_python_parser/src/lib.rs

```diff
@@ -582,6 +624,11 @@ pub enum Mode {
     /// The code consists of a single expression.
     Expression,
 
+    /// The code is parsed as if it is parenthesized. The code does not need to contain parenthesis
+    /// at the beginning. And it is assumed by the lexer that there is a parenthesis.
+    /// This allows the lexer to allow certain indentations that are not valid in a python file.
+    ParenthesizedExpression,
```

Comment:
I've suggested some improvements and I don't think we should specify the internal details about implementation within the lexer.

```suggestion
    /// The code consists of a single expression and is parsed as if it is parenthesized. The parentheses themselves aren't required.
    /// This allows for having valid multiline expression without the need of parentheses
    /// and is specifically useful for parsing string annotations.
    ParenthesizedExpression,
```